etcdserver/api/v3rpc/util.go

```diff
@@ -49,7 +49,7 @@ func togRPCError(err error) error {
 	case etcdserver.ErrTooManyRequests:
 		return rpctypes.ErrTooManyRequests
 
-	case etcdserver.ErrNoLeader:
+	case etcdserver.ErrNoLeader, rpctypes.ErrNoLeader: // ErrNoLeader from leasehttp
```

Comment:
Oops, OK, it's hidden through lessor, but it's still a bald error type. Putting the rpctypes error in the case still seems like it's violating some assumed design principle since it sticks out as different. Likewise, `leasehttp` has to import another package to get at the error . Perhaps what belongs here is a new error like `lease.ErrLessorUnavailable`?